### PR TITLE
OlmoEarth Embeddings using Google Earth Engine data

### PR DESCRIPTION
**Pull request by ivanzvonkov**
## Purpose
This PR adds the code necessary to produce a Togo cropland map using OlmoEarth embeddings and Google Earth Engine

## Contents

**generate_embeddings.ipynb** 
One-stop shop Colab notebook for generating embeddings with OlmoEarth and Google Earth Engine. Inference can be done directly in the notebook or by deploying a Docker container.

**gee_create_OlmoEarth_cropland.js**
Google Earth Engine script for creating a Togo cropland using OlmoEarth embeddings in Cloud Storage.

**gee_OlmoEarth_cropland_eval.js**
Google Earth Engine script for evaluating the OlmoEarth cropland map and comparing with other cropland maps in Togo.

**deploy.sh**
Script to build and deploy OlmoEarth embedding generation docker container to Google Cloud.

**Dockerfile**
Builds docker container for running job.py

**job.py**
Python script for running inference and generating embeddings using OlmoEarth and two Google Cloud Storage buckets. Expects Google Earth Engine data to already be exported using Colab notebook.

---

> [!NOTE]
> Introduces a Dockerized inference job, GEE scripts, and a Colab workflow to generate OlmoEarth embeddings and produce/evaluate a Togo cropland map using GCS buckets.
> 
> - **Inference Pipeline**:
>   - Adds `scripts/inference/job.py` to load pretrained encoder, normalize inputs, run tiled inference, and write 192-d embeddings to GeoTIFFs in `OUT_BUCKET`.
>   - Introduces `scripts/inference/Dockerfile` to containerize the job with required deps and env vars (`GCLOUD_PROJECT`, `IN_BUCKET`, `OUT_BUCKET`).
> - **Google Earth Engine (GEE)**:
>   - Adds `scripts/inference/gee_create_OlmoEarth_cropland.js` to mosaic embeddings from GCS, train RF on labeled points, and export cropland probabilities as an asset.
>   - Adds `scripts/inference/gee_OlmoEarth_cropland_eval.js` to evaluate cropland map against WorldCover and GLAD using test points.
> - **Notebook**:
>   - Adds `scripts/inference/generate_embeddings.ipynb` for end-to-end setup: export GEE tiles to GCS, run OlmoEarth inference, and visualize data.
> - **Deployment**:
>   - Adds `scripts/inference/deploy.sh` to fetch model weights, build and push the container to Artifact Registry.
> - **Misc**:
>   - Adds `scripts/inference/.gitignore` to ignore local model weights.
>   - Adds `scripts/inference/README.md` documenting components and workflow.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5f6a81b39d368531dd9a07fc018b2fa5141e5d60. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: scripts/inference/job.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        profile.update(count=EMBEDDINGS_SIZE, dtype="float32", compress="deflate", bigtiff="YES")
+
+        with rio.open("out.tif", "w", **profile) as dst:
+            for y in range(0, height, INFERENCE_WINDOW_SIZE):
+                for x in range(0, width, INFERENCE_WINDOW_SIZE):
+                    win = Window(x, y, min(INFERENCE_WINDOW_SIZE, width - x), min(INFERENCE_WINDOW_SIZE, height - y))
+                    data = src.read(window=win)
+                    masked_sample = prepare_masked_olmo_earth_sample(data, bands, timestamps, device)
+                    with torch.no_grad():
+                        preds = model(masked_sample, patch_size=1)
```
Comment (Hgherzog) [inline scripts/inference/job.py]: For inference, we should set fast_pass=True in order to use flash attention and avoid unneeded grad sync during training.

Comment (ivanzvonkov) [reply scripts/inference/job.py]: Added!
